src/ansi-c/c_typecheck_base.cpp

```diff
@@ -674,6 +674,28 @@ void c_typecheck_baset::apply_asm_label(
   }
 }
 
+/// Recursively set function name in source locations of expression
```

Comment:
Would it not be better to set the function name when parsing the contract?

Reply:
Remi and I were under the impression that the function name doesn't exist or isn't readily available at parsing time.  This is because the function doesn't fully exist until the body has been parsed.  If it is possible during parsing, that would be a better solution, but I didn't see a way to make it happen.

Reply:
You've got the function name once the declarator has been seen; this is well before the contract.

Reply:
@kroening I've now changed this in the parser and have removed changes from type-checking.  The parser will now set the function in source location, but only for top-level functions.  I think this change results in the correct behavior and excludes problems introduced by function pointers.